src/phpDebug.ts

```diff
@@ -898,20 +898,19 @@ class PhpDebugSession extends vscode.DebugSession {
         try {
             const connection = this._connections.get(args.threadId)
             if (!connection) {
-                throw new Error('Unknown thread ID ' + args.threadId)
+                return this.sendErrorResponse(response, new Error('Unknown thread ID ' + args.threadId))
             }
+            response.body = {
+                allThreadsContinued: false,
+            }
+            this.sendResponse(response)
             xdebugResponse = await connection.sendRunCommand()
         } catch (error) {
-            this.sendErrorResponse(response, error)
```

Comment:
If this is removed, it would be good to at least log the error to the user to the debug console.

Reply:
I have refactored the code to handle all error scenarios. Those before sending a response and those after already sending a response.

I also written a generalized helper method, since all step calls (continueRequest, nextRequest, stepInRequest and stepOutRequest) are basically the same.
I have not committed it, as it does introduce some indirection and makes the code a bit harder to follow. Here's the function, we can still use it if you think it's worth it.

```typescript
    protected async stepRequestHelper<T extends VSCodeDebugProtocol.Response>(name: string, threadId: number, response: T, op: (connection: xdebug.Connection)=>Promise<xdebug.StatusResponse>) {
        let connection: xdebug.Connection | undefined
        try {
            connection = this._connections.get(threadId)
            if (!connection) {
                return this.sendErrorResponse(response, new Error('Unknown thread ID ' + threadId))
            }
            this.sendResponse(response)
        } catch (error) {
            this.sendErrorResponse(response, error)
            return
        }
        try {
            const xdebugResponse = await op(connection)
            this._checkStatus(xdebugResponse)
        } catch (error) {
            this.sendEvent(new vscode.OutputEvent(`${name} thread Id ${threadId} error: ${error.message}\n`))
        }
    }
```